savedstateflow-hilt/build.gradle

```diff
@@ -0,0 +1,56 @@
+plugins {
+    id 'com.android.library'
+    id 'org.jetbrains.kotlin.android'
+    id 'kotlin-kapt'
+    id 'dagger.hilt.android.plugin'
+}
+
+ext {
+    PUBLISH_GROUP_ID = "$publish_group_id"
+//    PUBLISH_VERSION = "$publish_version"
+    PUBLISH_VERSION = "0.1-alpha01"
+    PUBLISH_ARTIFACT_ID = 'saved-state-flow-hilt'
+}
+
+android {
+    compileSdk compile_sdk
+
+    defaultConfig {
+        minSdk min_sdk
+        targetSdk target_sdk
+
+        testInstrumentationRunner "androidx.test.runner.AndroidJUnitRunner"
+        consumerProguardFiles "consumer-rules.pro"
+    }
+
+    buildTypes {
+        release {
+            minifyEnabled false
+            proguardFiles getDefaultProguardFile('proguard-android-optimize.txt'), 'proguard-rules.pro'
+        }
+    }
+    compileOptions {
+        sourceCompatibility JavaVersion.VERSION_1_8
+        targetCompatibility JavaVersion.VERSION_1_8
+    }
+    kotlinOptions {
+        jvmTarget = '1.8'
+    }
+}
+
+dependencies {
+    api(project(':savedstateflow'))
+    implementation "com.google.dagger:hilt-android:$hilt_version"
+    kapt "com.google.dagger:hilt-compiler:$hilt_version"
+
+    implementation "androidx.appcompat:appcompat:$appcompat_version"
+    implementation "androidx.lifecycle:lifecycle-livedata-ktx:$livedata_ktx_version"
+    implementation "androidx.lifecycle:lifecycle-viewmodel-ktx:$viewmodel_ktx_version"
+
+    testImplementation "junit:junit:$junit_version"
+    testImplementation "io.mockk:mockk:$mockk_version"
+    testImplementation "org.jetbrains.kotlinx:kotlinx-coroutines-test:$coroutines_version"
+    testImplementation "app.cash.turbine:turbine:$turbine_version"
```

Comment:
audit which imports here are actually needed

Reply:
You don't need `androidx.appcompat`, `livedata-ktx` & `viewmodel-ktx` since you are writing code only related to DI for which hilt got you covered :)

Reply:
Thank you, I might need the appcompat because I thought of another use case for asssisted injection with a `SavedStateFlowHandle` in case somebody needed to inject something at run time. So will be adding these methods with some documentation. original [idea](https://github.com/google/dagger/issues/2287#issuecomment-762249856) for these. 

```kotlin
/**
 * Get a reference to a [ViewModel] that is assisted injected with a [SavedStateFlowHandle] that is scoped
 * to the [Fragment]
 */
inline fun <reified T : ViewModel> Fragment.assistedViewModel(
    crossinline viewModelProducer: (SavedStateFlowHandle) -> T
): Lazy<T> = viewModels {
    object : AbstractSavedStateViewModelFactory(this, arguments) {
        override fun <T : ViewModel> create(
            key: String,
            modelClass: Class<T>,
            handle: SavedStateHandle
        ) = viewModelProducer(handle.toSavedStateFlowHandle()) as T
    }
}

/**
 * Get a reference to a [ViewModel] that is assisted injected with a [SavedStateFlowHandle] that is scoped
 * to its Activity
 */
inline fun <reified T : ViewModel> Fragment.assistedActivityViewModel(
    crossinline viewModelProducer: (SavedStateFlowHandle) -> T
): Lazy<T> = activityViewModels {
    object : AbstractSavedStateViewModelFactory(this, arguments) {
        override fun <T : ViewModel> create(
            key: String,
            modelClass: Class<T>,
            handle: SavedStateHandle
        ) = viewModelProducer(handle.toSavedStateFlowHandle()) as T
    }
}

/**
 * Get a reference to a [ViewModel] that is assisted injected with a [SavedStateFlowHandle] that is scoped
 * to a [FragmentActivity]
 */
inline fun <reified T : ViewModel> FragmentActivity.assistedViewModel(
    crossinline viewModelProducer: (SavedStateFlowHandle) -> T
): Lazy<T> = viewModels {
    object : AbstractSavedStateViewModelFactory(this, intent.extras) {
        override fun <T : ViewModel> create(
            key: String,
            modelClass: Class<T>,
            handle: SavedStateHandle
        ) = viewModelProducer(handle.toSavedStateFlowHandle()) as T
    }
}
```